lib/AST/ASTImporter.cpp

```diff
@@ -2747,11 +2747,8 @@ Decl *ASTNodeImporter::VisitFriendDecl(FriendDecl *D) {
   FriendDecl::FriendUnion ToFU;
   if (NamedDecl *FriendD = D->getFriendDecl()) {
     auto ToFriendD = cast_or_null<NamedDecl>(Importer.Import(FriendD));
-    if (ToFriendD && FriendD->getFriendObjectKind())
-      ToFriendD->setObjectOfFriendDecl();
```

Comment:
So you no longer need this to be called on the target? Can IdentifierNamespace equality substitute for this?

Reply:
Exactly.
`setObjectOfFriendDecl` sets the member `IdentifierNamespace` to value which represent that the `Decl` is a friend.
But that is already handled with `ToD->IdentifierNamespace = FromD->IdentifierNamespace;` in `Import(Decl* FromD)`.